ui/components/src/TimeSeriesTooltip/TooltipHeader.tsx

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Box, Divider, Typography, Stack, Switch } from '@mui/material';
+import Pin from 'mdi-material-ui/Pin';
+import PinOutline from 'mdi-material-ui/PinOutline';
+import React from 'react';
+import { useTimeZone } from '../context/TimeZoneProvider';
+import { TOOLTIP_BG_COLOR_FALLBACK, TOOLTIP_MAX_WIDTH } from './tooltip-model';
+
+export interface TooltipHeaderProps {
+  seriesTimeMs: number;
+  totalSeries: number;
+  isTooltipPinned: boolean;
+  showAllSeries: boolean;
+  onShowAllClick?: (checked: boolean) => void;
+  onUnpinClick?: () => void;
+}
+
+export const TooltipHeader = React.memo(function TooltipHeader({
```

Comment:
Dumb question, I often see `React.******` imports, isn't it better to only import the function or variable "******" instead of importing all the "React" namespace?

Reply:
Yeah no good reason not to `import { memo } from 'react';` here instead, I'll update this